ADAL/src/ui/ios/UIAlertView+Additions.m

```diff
@@ -1,51 +1,63 @@
 #import <objc/runtime.h>
 #import "ADWebAuthController.h"
 #import "ADALFrameworkUtils.h"
+#import "ADAppExtensionUtil.h"
 
 @implementation UIAlertView (Additions)
 
-static const char *HANDLER_KEY = "com.microsoft.adal.alertviewHandler";
+static const char* HANDLER_KEY = "com.microsoft.adal.alertviewHandler";
 
-static UIAlertView *alert;
+static UIAlertView* alert;
 
-+ (void)presentCredentialAlert:(void (^)(NSUInteger))handler {
-    
-#if !ADAL_EXTENSION_SAFE
-    NSBundle* bundle = [ADALFrameworkUtils frameworkBundle];
-    if (!bundle)
++ (void)presentCredentialAlert:(void (^)(NSUInteger))handler
+{
+    if (![ADAppExtensionUtil isExecutingInAppExtension])
     {
-        bundle = [NSBundle mainBundle];
+        NSBundle* bundle = [ADALFrameworkUtils frameworkBundle];
+        
+        if (!bundle)
+        {
+            bundle = [NSBundle mainBundle];
+        }
+
+        alert = [[UIAlertView alloc] initWithFrame:CGRectZero];
+        alert.title = NSLocalizedStringFromTableInBundle(@"Enter your credentials", nil, bundle, nil);
+        alert.alertViewStyle = UIAlertViewStyleLoginAndPasswordInput;
+        
+        [alert addButtonWithTitle:NSLocalizedStringFromTableInBundle(@"Cancel", nil, bundle, nil)];
+        [alert addButtonWithTitle:NSLocalizedStringFromTableInBundle(@"Login", nil, bundle, nil)];
+        [alert setCancelButtonIndex:0];
```

Comment:
You need to get a server to send you an NTLM challenge. The only way I have to do it involves using a MS STS while on corpnet and your corpnet account. I'll try to repro and send you directions.